src/lib389/lib389/__init__.py

```diff
@@ -3484,3 +3485,17 @@ def restore_online(self, archive, db_type=None):
         task.create(properties=task_properties)
 
         return task
+
+    def is_rootdn_bound(self):
+        # Return True if root DN is authenticated for this DirSrv instance
+        if self.binddn is None and get_user_is_root():
```

Comment:
This code is fine in dsidm case but I think we should improve it to avoid future trouble if someone reuse the function:
   only the fact that is instance is connected ensure that "bind dn is None
      " means that "ldapi was successfully used".
    so I would add a test to return False if  self.state != DIRSRV_STATE_ONLINE
    otherwise root user specifying no bind dn to remote instance may be perceived as root_dn bound.
Except that minor point, LGTM

Reply:
fixed